src/circuit/gadget/poseidon/pow5.rs

```diff
@@ -715,49 +719,49 @@ mod tests {
     #[test]
     fn poseidon_permute() {
         let k = 6;
-        let circuit = PermuteCircuit {};
+        let circuit = PermuteCircuit::<OrchardNullifier, 3, 2>(PhantomData);
         let prover = MockProver::run(k, &circuit, vec![]).unwrap();
         assert_eq!(prover.verify(), Ok(()))
     }
 
-    #[derive(Default)]
-    struct HashCircuit {
+    struct HashCircuit<S: Spec<Fp, WIDTH, RATE>, const WIDTH: usize, const RATE: usize> {
         message: Option<[Fp; 2]>,
```

Comment:
This should be:
```suggestion
        message: Option<[Fp; L]>,
```
for a new `const L: usize`, since it corresponds to the constant length of the message being hashed.